airbyte-integrations/connectors/destination-snowflake/src/test-integration/kotlin/io/airbyte/integrations/destination/snowflake/operation/SnowflakeStagingClientIntegrationTest.kt

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.snowflake.operation
+
+import com.fasterxml.jackson.databind.JsonNode
+import io.airbyte.cdk.db.jdbc.JdbcDatabase
+import io.airbyte.cdk.integrations.destination.record_buffer.FileBuffer
+import io.airbyte.cdk.integrations.destination.s3.csv.CsvSerializedBuffer
+import io.airbyte.cdk.integrations.destination.s3.csv.CsvSheetGenerator
+import io.airbyte.commons.json.Jsons
+import io.airbyte.commons.string.Strings
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId
+import io.airbyte.integrations.destination.snowflake.OssCloudEnvVarConsts
+import io.airbyte.integrations.destination.snowflake.SnowflakeDatabaseUtils
+import io.airbyte.protocol.models.v0.AirbyteRecordMessage
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.time.Instant
+import java.util.*
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class SnowflakeStagingClientIntegrationTest {
+
+    private lateinit var stagingClient: SnowflakeStagingClient
+    // Not using lateinit to keep spotBugs happy
+    // since these vars are referenced within the setup
+    // and generated bytecode as if non-null check
+    private var namespace: String = ""
+    private var tablename: String = ""
+
+    private lateinit var stageName: String
+    private val config =
+        Jsons.deserialize(Files.readString(Paths.get("secrets/1s1t_internal_staging_config.json")))
+    private val datasource =
+        SnowflakeDatabaseUtils.createDataSource(config, OssCloudEnvVarConsts.AIRBYTE_OSS)
+    private val database: JdbcDatabase = SnowflakeDatabaseUtils.getDatabase(datasource)
+    // Intentionally not using actual columns, since the staging client should be agnostic of these
```

Comment:
random thought: do we eventually want to explicitly specify column names in our `COPY` command? 😅

Reply:
I guess so.. it requires `COPY INTO .. select filename#colindex` or something i think right ?

Reply:
something like that :shrug: not sure about the exact syntax, presumably there's some way to specify "please load into _exactly these columns_ in the target table"

Reply:
which I guess is actually slightly different from what you have (i.e. "read these columns out of the file" vs "write to these columns in the table")